server/src/main/java/org/candlepin/async/JobManager.java

```diff
@@ -1440,6 +1457,96 @@ private AsyncJobStatus processJobFailure(AsyncJobStatus status, EventSink eventS
         return status;
     }
 
+    /**
+     * Cancels the job associated with the specified job ID. If no such job status could be found,
+     * this method returns null.
+     *
+     * @param jobId
+     *  the ID of the job to cancel
+     *
+     * @throws IllegalStateException
+     *  if the job is already in a terminal state
+     *
+     * @return
+     *  the updated job status associated with the canceled job, or null if no such job status
+     *  could be found
+     */
+    @Transactional
+    public synchronized AsyncJobStatus cancelJob(String jobId) {
+        if (jobId == null || jobId.isEmpty()) {
+            throw new IllegalArgumentException("jobId is null or empty");
+        }
+
+        AsyncJobStatus status = this.jobCurator.lockAndLoad(jobId);
```

Comment:
Do we need a lock? Could we use an optimistic "update where" & react appropriately if the update fails to update any rows?